### PR TITLE
zsh: move session variables export to zshrc

### PR DESCRIPTION
**Pull request by uvNikita**
Unlike .zshenv, .zshrc file is sourced only by interactive shells.

According to http://zsh.sourceforge.net/Intro/intro_3.html, `.zlogin` and `.zprofile`
shouldn't be used to set environment variables.

[Here](http://zsh.sourceforge.net/Contrib/startup/std/zshenv) it also says:

```
.zshenv is sourced on ALL invocations of the shell, unless the -f option is
set.  It should NOT normally contain commands to set the command search path,
or other common environment variables unless you really know what you're
doing.
```

There is also example [.zshenv](http://zsh.sourceforge.net/Contrib/startup/users/debbiep/dot.zshenv) file where it's stated that
`Only environment variables that must be set for a noninteractive shell are set here`.

Since `programs.zsh.sessionVariables` option will probably be used with interactive shell in mind, `.zshrc` seems to be more suitable place to set variables.

Relevant to #183, but I'm not sure if the patch fixes the issue.

### CODE REVIEW

File: modules/programs/zsh.nix [hunk 1]
```diff
@@ -240,20 +238,46 @@ in
...
+    })
+
+    (mkIf (cfg.logoutExtra != "") {
+      home.file."${relToDotDir ".zlogout"}".text = cfg.logoutExtra;
+    })
+
+    (mkIf cfg.oh-my-zsh.enable {
+      home.file."${relToDotDir ".zshenv"}".text = ''
+        ZSH="${pkgs.oh-my-zsh}/share/oh-my-zsh";
+        ZSH_CACHE_DIR="''${XDG_CACHE_HOME:-''$HOME/.cache}/oh-my-zsh";
```
Comment (Infinisil) [inline modules/programs/zsh.nix]: Why are these (and `ZDOTDIR`) now written directly instead of using sessionVariables?

Comment (uvNikita) [reply modules/programs/zsh.nix]: Like @rycee pointed out, it seems that ZSH, ZSH_CACHE_DIR, and ZDOTDIR are used only by zsh process itself and therefore don't have to be exported (which would happen if we would use sessionVariables). I'm not 100% sure, so please correct me if I'm wrong.

Comment (Infinisil) [reply modules/programs/zsh.nix]: Yeah makes sense, after thinking about it for a bit I can't find anything that wouldn't work because of this.

**Comment by uvNikita** [comment]
Rebased into master: https://github.com/rycee/home-manager/commit/fa6f697dbbe06b47b17057d93dffdfcd6f39cbbc.

**Pull request closed without merging**